activation/activation_test.go

```diff
@@ -1427,3 +1427,53 @@ func TestWaitingToBuildNipostChallengeWithJitter(t *testing.T) {
 		require.Less(t, deadline, time.Now())
 	})
 }
+
+// Test if GetPositioningAtx disregards ATXs with invalid POST in their chain.
+// It should pick an ATX with valid POST even though it's a lower height.
+func TestGetPositioningAtxPicksAtxWithValidChain(t *testing.T) {
+	tab := newTestBuilder(t)
+
+	// Invalid chain with high height
+	sigInvalid, err := signing.NewEdSigner()
+	require.NoError(t, err)
+	ch := newChallenge(1, types.EmptyATXID, tab.goldenATXID, postGenesisEpoch, &tab.goldenATXID)
+	nipostData := newNIPostWithChallenge(t, types.HexToHash32(""), []byte("0"))
+	invalidAtx := newAtx(t, sigInvalid, ch, nipostData.NIPost, 2, types.Address{})
+	require.NoError(t, SignAndFinalizeAtx(sigInvalid, invalidAtx))
+	vInvalidAtx, err := invalidAtx.Verify(0, 100)
+	require.NoError(t, err)
+	require.NoError(t, atxs.Add(tab.cdb, vInvalidAtx))
+
+	// Valid chain with lower height
+	sigValid, err := signing.NewEdSigner()
+	require.NoError(t, err)
+	ch = newChallenge(1, types.EmptyATXID, tab.goldenATXID, postGenesisEpoch, &tab.goldenATXID)
+	nipostData = newNIPostWithChallenge(t, types.HexToHash32(""), []byte("1"))
+	validAtx := newAtx(t, sigValid, ch, nipostData.NIPost, 2, types.Address{})
+	require.NoError(t, SignAndFinalizeAtx(sigValid, validAtx))
+	vValidAtx, err := validAtx.Verify(0, 1)
+	require.NoError(t, err)
+	require.NoError(t, atxs.Add(tab.cdb, vValidAtx))
+
+	tab.mValidator.EXPECT().
+		VerifyChain(gomock.Any(), invalidAtx.ID(), tab.goldenATXID, gomock.Any()).
+		Return(errors.New(""))
+	tab.mValidator.EXPECT().
+		VerifyChain(gomock.Any(), validAtx.ID(), tab.goldenATXID, gomock.Any())
+
+	posAtxID, err := tab.getPositioningAtx(context.Background())
+	require.NoError(t, err)
+	require.Equal(t, posAtxID, vValidAtx.ID())
+}
+
+func TestGetPositioningAtxDbFailed(t *testing.T) {
+	tab := newTestBuilder(t)
+	db := datastoremocks.NewMockExecutor(gomock.NewController(t))
+	tab.Builder.cdb = datastore.NewCachedDB(db, logtest.New(t))
+	expected := errors.New("db error")
+	db.EXPECT().Exec(gomock.Any(), gomock.Any(), gomock.Any()).Return(0, expected)
+
+	none, err := tab.getPositioningAtx(context.Background())
+	require.ErrorIs(t, err, expected)
+	require.Equal(t, types.ATXID{}, none)
+}
```

Comment:
I'm not understanding this test - aren't you just testing here that the `datastore.CachedDB` returns an error when the DB it uses returns an error? Why is this here and not in the `datastore` package?